services/api/kalvi/api/views/auth.py

```diff
@@ -1,15 +1,116 @@
 from rest_framework.views import APIView
 from rest_framework.response import Response
+from rest_framework import status
+from db.serializers import SignUpEndPointSerializer, SignInEndPointSerializer, UserProfileSerialiser, UserChangePasswordSerialiser, SendPasswordResetEmailSerializer, UserPasswordResetSerializer
+from django.contrib.auth import authenticate
+from db.renderer import UserRenderer
+from rest_framework_simplejwt.tokens import RefreshToken
+from rest_framework.permissions import IsAuthenticated, AllowAny
+from django.utils.encoding import smart_str
+from django.utils.http import urlsafe_base64_decode
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
+from db.models import User
 
+# Generate stateless token for User
+def get_tokens_for_user(user):
+  refresh = RefreshToken.for_user(user)
+  return {
+      'refresh': str(refresh),
+      'access': str(refresh.access_token),
+  }
 
+#viewset class for sign up
 class SignUpEndPoint(APIView):
-    def post(self, request):
-        return Response("SIGNUP ENDPOINT")
+    renderer_classes = [UserRenderer]
+    permission_classes = [AllowAny]  # Allow signup for unregistered users
+    def post(self, request, format=None):
+        serializer = SignUpEndPointSerializer(data=request.data)
+        if serializer.is_valid(raise_exception=True):
+            try:
+                user = serializer.save()
+                token = get_tokens_for_user(user)
+                return Response({'token': token, 'message': 'user registered'}, status=status.HTTP_201_CREATED)
+            except Exception as e:  # Catch potential errors during user creation
+                return Response({'error': str(e)}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+        else:
+            return Response(ValidationError, status=status.HTTP_400_BAD_REQUEST)
 
-    def get(self, request):
-        return Response("SIGNUP ENDPOINT")
+#viewset class for sign in 
+class SignInEndPoint(APIView):
+    renderer_classes = [UserRenderer]
+    def post(self, request, format=None):
+        serializer = SignInEndPointSerializer(data=request.data)
+        try:    
+            serializer.is_valid(raise_exception=True)
+            email = serializer.validated_data['email']
+            password = serializer.validated_data['password']
+            user = authenticate(email=email, password=password)
+            if user is not None:
+                token = get_tokens_for_user(user)
+                return Response({'token' : token,'message' : 'Login Successful'}, status=status.HTTP_200_OK)
+            else:
+                #more specific error message to avoid revealing authentication mechanisms
+                error_message = "Invalid credentials, you need to register first."
+                return Response({'errors': error_message}, status=status.HTTP_401_UNAUTHORIZED)
+        except (ValidationError, PermissionError) as e:
```

Comment:
should 401 status be used for permission error?

Reply:
I guess it is catching if somebody is logging with unauthorized password. so i came up with 401 thats what cgat gpt suggested . are you confident should i change it?
<img width="585" alt="Screenshot 2024-02-21 134236" src="https://github.com/kalvilabs/kalvios/assets/157835366/af8f9072-8f21-48b1-bf68-ff69c47e909a">

Reply:
I'm closing it